src/frequenz/sdk/microgrid/metadata.py

```diff
@@ -0,0 +1,40 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Metadata that describes a microgrid."""
+
+from dataclasses import dataclass
+
+from timezonefinder import TimezoneFinder
+
+_timezone_finder = TimezoneFinder()
+
+
+@dataclass(frozen=True)
+class Location:
+    """Metadata for the location of microgrid."""
+
+    latitude: float
+    """The latitude of the microgrid."""
+
+    longitude: float
+    """The longitude of the microgrid."""
+
+    def get_timezone(self) -> str | None:
+        """Get the timezone string of the microgrid.
+
+        Returns:
+            the timezone string of the microgrid if found, None otherwise.
+        """
+        return _timezone_finder.timezone_at(lat=self.latitude, lng=self.longitude)
```

Comment:
What about something like:

```suggestion
    _timezone: str | None = None

    def __post_init__(self) -> None:
        self._timezone = _timezone_finder.timezone_at(lat=self.latitude, lng=self.longitude)
        
    @property
    def timezone(self) -> str | None:
        """The timezone string for this location."""
        return self._timezone
```

?

Also, is there any way to get a [`timezone` object](https://docs.python.org/3/library/datetime.html#timezone-objects) instead (or too)? I guess that's what data scientist need the most...

Reply:
we could return a [ZoneInfo](https://docs.python.org/3/library/zoneinfo.html) instead that can be used with datetime objects

Reply:
I can implement `__post_init__` but I'm afraid `Location` class can no longer be frozen